app/org/sagebionetworks/bridge/hibernate/HibernateAccountDao.java

```diff
@@ -303,14 +315,20 @@ public void createAccountForMigration(Study study, Account account, String id, b
         try {
             hibernateHelper.create(hibernateAccount);
         } catch (ConcurrentModificationException ex) {
-            // account exists, but we don't have the userId, load the account
-            AccountId accountId = AccountId.forEmail(study.getIdentifier(), account.getEmail());
+            // Account can conflict because studyId + email or studyId + phone has been used for an 
+            // existing account. 
+            AccountId accountId = null;
+            if (account.getEmail() != null) {
+                accountId = AccountId.forEmail(study.getIdentifier(), account.getEmail());
+            } else if (account.getPhone() != null) {
+                accountId = AccountId.forPhone(study.getIdentifier(), account.getPhone());
+            }
```

Comment:
Do we need an else case for an impossible codepath? Or is a NullPointerException acceptable in this situation?

Reply:
If we got a concurrent modification exception, this account should really have an email or phone number... it seems fine to allow an NPE further along in the code path.